packages/fx-core/src/plugins/solution/fx-solution/v2/adaptor.ts

```diff
@@ -107,3 +107,36 @@ export class ProvisionContextAdapter extends BaseSolutionContextAdaptor {
     );
   }
 }
+
+export class CollaboratorContextAdapter extends BaseSolutionContextAdaptor {
+  constructor(params: Parameters<NonNullable<v2.SolutionPlugin["grantPermission"]>>) {
+    super();
+    const v2context: v2.Context = params[0];
+    const inputs: Inputs = params[1];
+    const envInfo = params[2];
+    const tokenProvider = params[3];
+    this.root = inputs.projectPath ?? "";
```

Comment:
What will happen when root is empty?

Reply:
root should never be empty here.  What I want here is just to cast away the null-ness of projectPath for simplicity. I'll change to throw error to enforce the fact the root will not be undefined.